src/App.tsx

```diff
@@ -15,36 +15,94 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabet = 'alphabet',
+  Length = 'length',
+}
+
+interface SortParams {
+  sortField: string;
+  isReverse: boolean;
+}
+
+const getPreparedGoods = (goods: string[], sorter: SortParams) => {
+  const { sortField, isReverse } = sorter;
+  const prepGoods = [...goods];
+
+  if (sortField === SortType.Alphabet) {
+    prepGoods.sort((a, b) => a.localeCompare(b));
+  }
+
+  if (sortField === SortType.Length) {
+    prepGoods.sort((a, b) => a.length - b.length);
+  }
+
+  if (isReverse) {
+    prepGoods.reverse();
+  }
+
+  return prepGoods;
+};
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReverse, setIsReverse] = useState(false);
+
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortField,
+    isReverse,
+  });
+
+  const handleReset = () => {
+    setIsReverse(false);
+    setSortField('');
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={`button is-info ${sortField === SortType.Alphabet ? '' : 'is-light'}`}
+          onClick={() => setSortField(SortType.Alphabet)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={`button is-success is-light" ${sortField === SortType.Length ? '' : 'is-light'}`}
```

Comment:
There is a typo in the className string. The className should not have a stray double quote (`"`) after `is-light`. It should be `className={`button is-success ${sortField === SortType.Length ? '' : 'is-light'}`}`.